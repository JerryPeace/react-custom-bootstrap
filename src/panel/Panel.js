import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classNames from 'classnames';
import bootstrapUtils from '../utils/bootstrapUtils';
import Mask from '../mask/Mask';

class Panel extends Component {

    static propTypes = {
        bsClass: PropTypes.string,
        bsStyle: PropTypes.string,
        header: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
        body: PropTypes.node,
        footer: PropTypes.node,
        mask: PropTypes.bool
    };

    static defaultProps = {
        bsClass: 'panel',
        bsStyle: 'default',
        mask: false
    };

    renderHeader() {
        let header = this.props.header;

        if (!header) {
            return null;
        }

        if (typeof header === 'string') {
            header = <h6 className={'panel-title'}>{header}</h6>;
        }
        return (
            <div className={'panel-heading'}>
                {header}
            </div>
        );
    }

    renderBody() {
        const {mask, body, children} = this.props;
        return (
            <div className={classNames('panel-body', {'maskContainer': mask})}>
                {body}
                {children}
                {mask ? <Mask /> : null}
            </div>
        );
    }

    renderFooter() {
        const footer = this.props.footer;

        return !footer ? null : (
            <div className={'panel-footer'}>{footer}</div>
        );
    }

    render() {
        const {
            body,
            bsClass,
            bsStyle,
            children,
            className,
            footer,
            header,
            mask,
            ...rest
        } = this.props;

        const bsCls = bootstrapUtils.getCls(this.props);

        return (
            <div {...rest} className={classNames(className, bsCls)} >
                {this.renderHeader()}
                {this.renderBody()}
                {this.renderFooter()}
            </div>
        );
    }
}

export default Panel;
