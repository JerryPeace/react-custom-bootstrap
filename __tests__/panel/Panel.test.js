import React from 'react';
import { mount } from 'enzyme';
import _ from 'lodash';
import Panel from 'panel/Panel';

describe('Panel', () => {
    test('render Panel default', () => {
        const panel = mount(
            <Panel>
                <span>test</span>
            </Panel>
        );
        expect(panel).toMatchSnapshot();
    });

    test('render Panel default with header text', () => {
        const panel = mount(
            <Panel header='test header'>
                <span>test</span>
            </Panel>
        );
        expect(panel).toMatchSnapshot();
    });

    test('render Panel default with header node', () => {
        const panel = mount(
            <Panel header={(<div>HEADER</div>)}>
                <span>test</span>
            </Panel>
        );
        expect(panel).toMatchSnapshot();
    });

    test('render Panel default with footer', () => {
        const panel = mount(
            <Panel footer={(<div>FOOTER</div>)}>
                <span>test</span>
            </Panel>
        );
        expect(panel).toMatchSnapshot();
    });

    test('render Panel default with mask', () => {
        const panel = mount(
            <Panel mask>
                <span>test</span>
            </Panel>
        );
        expect(panel).toMatchSnapshot();
    });

    test('render Panel default with all', () => {
        const panel = mount(
            <Panel mask header={(<div>HEADER</div>)} footer={(<div>FOOTER</div>)}>
                <span>test</span>
            </Panel>
        );
        expect(panel).toMatchSnapshot();
    });
});
