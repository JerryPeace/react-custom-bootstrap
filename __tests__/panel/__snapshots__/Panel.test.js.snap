// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel render Panel default 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  mask={false}
>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body"
    >
      <span>
        test
      </span>
    </div>
  </div>
</Panel>
`;

exports[`Panel render Panel default with all 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  footer={
    <div>
      FOOTER
    </div>
  }
  header={
    <div>
      HEADER
    </div>
  }
  mask={true}
>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <div>
        HEADER
      </div>
    </div>
    <div
      className="panel-body maskContainer"
    >
      <span>
        test
      </span>
      <Mask
        loaderProps={
          Object {
            "bsSize": "md",
          }
        }
        show={true}
      >
        <div
          className="tm-mask"
        >
          <div />
        </div>
      </Mask>
    </div>
    <div
      className="panel-footer"
    >
      <div>
        FOOTER
      </div>
    </div>
  </div>
</Panel>
`;

exports[`Panel render Panel default with footer 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  footer={
    <div>
      FOOTER
    </div>
  }
  mask={false}
>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body"
    >
      <span>
        test
      </span>
    </div>
    <div
      className="panel-footer"
    >
      <div>
        FOOTER
      </div>
    </div>
  </div>
</Panel>
`;

exports[`Panel render Panel default with header node 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  header={
    <div>
      HEADER
    </div>
  }
  mask={false}
>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <div>
        HEADER
      </div>
    </div>
    <div
      className="panel-body"
    >
      <span>
        test
      </span>
    </div>
  </div>
</Panel>
`;

exports[`Panel render Panel default with header text 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  header="test header"
  mask={false}
>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h6
        className="panel-title"
      >
        test header
      </h6>
    </div>
    <div
      className="panel-body"
    >
      <span>
        test
      </span>
    </div>
  </div>
</Panel>
`;

exports[`Panel render Panel default with mask 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  mask={true}
>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body maskContainer"
    >
      <span>
        test
      </span>
      <Mask
        loaderProps={
          Object {
            "bsSize": "md",
          }
        }
        show={true}
      >
        <div
          className="tm-mask"
        >
          <div />
        </div>
      </Mask>
    </div>
  </div>
</Panel>
`;
